src/sparseml/recipe_template/utils.py

```diff
@@ -0,0 +1,144 @@
+        variables defined for this modifier, defaults to "mod"
+    :param fields: An Optional dictionary mapping modifier attribute names to their
+        corresponding values
+    """
+
+    def __init__(
+        self,
+        modifier: Modifier,
+        modifier_name: str = "mod",
+        fields: Optional[Dict[str, Any]] = None,
+    ):
+        self.modifier = modifier
+        self.modifier_name = modifier_name
+        self.modifier_builder = ModifierYAMLBuilder(modifier)
+        self.fields = fields or {}
+        self.__modifier_recipe_variables = {}
+        self.update(updated_fields=self.fields)
+
+    def update(self, updated_fields: Optional[Dict[str, Any]] = None):
+        """
+        Utility method to update the fields for current modifier
+
+        :param updated_fields: An Optional dictionary mapping modifier attribute names
+            to their corresponding values
+        """
+        for key, value in updated_fields.items():
+            variable_name = f"{self.modifier_name}_{key}"
+            setattr(self.modifier_builder, key, f"eval({variable_name})")
+            self.__modifier_recipe_variables[variable_name] = updated_fields[key]
+
+    @property
+    def modifier_recipe_variables(self) -> Dict[str, Any]:
+        """
+        :returns: A dict mapping b/w recipe variable names --> values
+            for the current object
+        """
+        return self.__modifier_recipe_variables
+
+
+# PRUNING MODIFIERS INFO
+
+_PRUNING_MODIFIER_INFO_REGISTRY = {
+    "false": None,
+    "true": ModifierBuildInfo(
+        modifier=MagnitudePruningModifier,
+        modifier_name="pruning",
+        fields={
+            "init_sparsity": 0.05,
+            "final_sparsity": 0.8,
+            "start_epoch": 0.0,
+            "end_epoch": 10.0,
+            "update_frequency": 1.0,
+            "params": "__ALL_PRUNABLE__",
+            "leave_enabled": True,
+            "inter_func": "cubic",
+            "mask_type": "unstructured",
+        },
+    ),
+    "acdc": ModifierBuildInfo(
+        modifier=ACDCPruningModifier,
+        modifier_name="pruning",
+        fields={
+            "compression_sparsity": 0.9,
+            "start_epoch": 0,
+            "end_epoch": 100,
+            "update_frequency": 5,
+            "params": "__ALL_PRUNABLE__",
+            "global_sparsity": True,
+        },
+    ),
+    "gmp": ModifierBuildInfo(
+        modifier=GMPruningModifier,
+        modifier_name="pruning",
+        fields={
+            "init_sparsity": 0.05,
+            "final_sparsity": 0.8,
+            "start_epoch": 0.0,
+            "end_epoch": 10.0,
+            "update_frequency": 1.0,
+            "params": ["re:.*weight"],
+            "leave_enabled": True,
+            "inter_func": "cubic",
+            "mask_type": "unstructured",
+        },
+    ),
+}
+
+# QUANTIZATION MODIFIERS INFO
+
+_QUANTIZATION_MODIFIER_INFO_REGISTRY = {
+    "false": None,
+    "true": ModifierBuildInfo(
```

Comment:
Can we come up with a better name for this that's more specified to what it does? Could totally see adding other options for this in the future

Reply:
+1